src/main/scala/workflow/graph/DefaultOptimizer.scala

```diff
@@ -0,0 +1,11 @@
+package workflow.graph
+
+/**
+ * The default Pipeline optimizer used when executing pipelines.
+ */
+object DefaultOptimizer extends Optimizer {
+  protected val batches: Seq[Batch] =
+    Batch("Load Saved State", Once, ExtractSaveablePrefixes, SavedStateLoadRule, DanglingNodeRemovalRule) ::
```

Comment:
If we just combine ExtractSavablePrefixes and SavedStateLoadRule into a single rule do we eliminate the need for prefixes in other rules?